lib/motions/move-to-definition-motion.coffee

```diff
@@ -0,0 +1,40 @@
+_ = require 'underscore-plus'
+{Input} = require '../view-models/view-model'
+{SearchCurrentWord} = require './search-motion'
+
+module.exports =
+class MoveToDefinition extends SearchCurrentWord
+  constructor: (@editor, @vimState) ->
+    super(@editor, @vimState)
+
+    # FIXME: This must depend on the current language
+    # NOTE: The only modification here is the addition of the $
+    defaultIsKeyword = "[@$a-zA-Z0-9_\-]+"
+    userIsKeyword = atom.config.get('vim-mode.iskeyword')
+    @keywordRegex = new RegExp(userIsKeyword or defaultIsKeyword)
+
+    searchString = @getCurrentWordMatch()
+    @input = new Input(searchString)
```

Comment:
Very good, I can answer both questions together: I would suggest that we treat, for now, languages with variables starting with `$` as a user setting: in the tests with `$`, do a 
```
beforeEach -> 
  # a user may choose to include `$` as part of names:
  atom.config.set('vim-mode.iskeyword', "---the regex with $ in it---")
```
For example, I don't like having `@` as part of `*` searches, so I have, in my `init.coffee`, the command `atom.config.set('vim-mode.iskeyword', "[a-zA-Z0-9_]+")`

Yeah, it's very likely I think that `[[` and `gd` may mean `gD` will change, but for now, having a basic `gD` would be good.